librz/include/rz_util/rz_serialize.h

```diff
@@ -1,17 +1,106 @@
-/* rizin - LGPL - Copyright 2020 - thestr4ng3r */
+// SPDX-License-Identifier: LGPL-3.0-only
 
 #ifndef RZ_SERIALIZE_H
 #define RZ_SERIALIZE_H
 
 #include <rz_util/rz_json.h>
 #include <rz_list.h>
 
+/**
+ * \brief Detailed info about a (de)serialization result
+ *
+ * This is currently just a list of strings which may be warnings
+ * or detailed error messages.
+ */
 typedef RzList RzSerializeResultInfo;
+
 static inline RzSerializeResultInfo *rz_serialize_result_info_new(void) {
 	return rz_list_newf(free);
 }
+
 static inline void rz_serialize_result_info_free(RzSerializeResultInfo *info) {
 	rz_list_free(info);
 }
 
+// Common helpers for writing (de)serialization code
+
+/**
+ * \brief Push an error to the local RzSerializeResultInfo
+ * \param ... printf-style arguments to be pushed as the error
+ *
+ * To be used in deserialization functions with a parameter that is
+ * exactly `RzSerializeResultInfo *res`.
+ */
+#define RZ_SERIALIZE_ERR(...) \
```

Comment:
since this is now a global macro, would it be better to just pass `res` as argument instead of assuming there is a var named `res`?

Reply:
Agreed, better to make them explicit now.